drivers/bluetooth/hci/hci_stm32wba.c

```diff
@@ -0,0 +1,380 @@
+		data += sizeof(hdr);
+		len -= sizeof(hdr);
+	} else {
+		LOG_ERR("No available ACL buffers!");
+		return NULL;
+	}
+
+	if (ext_len != sys_le16_to_cpu(hdr.len)) {
+		LOG_ERR("ACL payload length is not correct");
+		net_buf_unref(buf);
+		return NULL;
+	}
+
+	net_buf_add_mem(buf, &hdr, sizeof(hdr));
+	buf_tailroom = net_buf_tailroom(buf);
+	if (buf_tailroom < len) {
+		LOG_ERR("Not enough space in buffer %zu/%zu", len, buf_tailroom);
+		net_buf_unref(buf);
+		return NULL;
+	}
+
+	LOG_DBG("ext_len %u", ext_len);
+	net_buf_add_mem(buf, ext_data, ext_len);
+
+	return buf;
+}
+
+static struct net_buf *treat_iso(uint8_t *data, size_t len,
+				 uint8_t *ext_data, uint16_t ext_len)
+{
+	struct bt_hci_iso_hdr hdr;
+	struct net_buf *buf;
+	size_t buf_tailroom;
+
+	if (len < sizeof(hdr)) {
+		LOG_ERR("Not enough data for ISO header");
+		return NULL;
+	}
+
+	buf = bt_buf_get_rx(BT_BUF_ISO_IN, K_NO_WAIT);
+	if (buf) {
+		memcpy((void *)&hdr, data, sizeof(hdr));
+		data += sizeof(hdr);
+		len -= sizeof(hdr);
+	} else {
+		LOG_ERR("No available ISO buffers!");
+		return NULL;
+	}
+
+	if (ext_len != bt_iso_hdr_len(sys_le16_to_cpu(hdr.len))) {
+		LOG_ERR("ISO payload length is not correct");
+		net_buf_unref(buf);
+		return NULL;
+	}
+
+	net_buf_add_mem(buf, &hdr, sizeof(hdr));
+	buf_tailroom = net_buf_tailroom(buf);
+	if (buf_tailroom < len) {
+		LOG_ERR("Not enough space in buffer %zu/%zu", len, buf_tailroom);
+		net_buf_unref(buf);
+		return NULL;
+	}
+
+	LOG_DBG("ext_len %zu", ext_len);
+	net_buf_add_mem(buf, ext_data, ext_len);
+
+	return buf;
+}
+
+static int receive_data(uint8_t *data, uint16_t len,
+			 uint8_t *ext_data, uint16_t ext_len)
+{
+	uint8_t pkt_indicator;
+	struct net_buf *buf;
+	int err = 0;
+
+	LOG_HEXDUMP_DBG(data, len, "host packet data:");
+	LOG_HEXDUMP_DBG(ext_data, ext_len, "host packet ext_data:");
+
+	pkt_indicator = *data++;
+	len -= sizeof(pkt_indicator);
+
+	switch (pkt_indicator) {
+	case HCI_EVT:
+		buf = treat_evt(data, len);
+		break;
+	case HCI_ACL:
+		buf = treat_acl(data, len + 1, ext_data, ext_len);
+		break;
+	case HCI_ISO:
+	case HCI_SCO:
+		buf = treat_iso(data, len + 1, ext_data, ext_len);
+		break;
+	default:
+		buf = NULL;
+		LOG_ERR("Unknown HCI type %u", pkt_indicator);
+	}
+
+	if (buf) {
+		bt_recv(buf);
+	} else {
+		err = -ENOMEM;
+		ll_state_busy = 1;
+	}
+
+	return err;
+}
+
+uint8_t BLECB_Indication(const uint8_t *data, uint16_t length,
+			 const uint8_t *ext_data, uint16_t ext_length)
+{
+	int ret = 0;
+	int err;
+
+	LOG_DBG("length: %d", length);
+	if (ext_length != 0) {
+		LOG_DBG("ext_length: %d", ext_length);
+	}
+
+	k_sem_take(&hci_sem, K_FOREVER);
+
+	err = receive_data((uint8_t *)data, (size_t)length - 1,
+			(uint8_t *)ext_data, (size_t)ext_length);
+
+	k_sem_give(&hci_sem);
+
+	HostStack_Process();
+
+	if (err) {
+		ret = 1;
+	}
+
+	return ret;
+}
+
+static int bt_hci_stm32wba_send(struct net_buf *buf)
+{
+	uint16_t event_length;
+	uint8_t pkt_indicator;
+	uint8_t tx_buffer[300];
```

Comment:
`300` seems a bit arbitrary. Perhaps at least add a define for it, though I'm surprised there's no `CFG_*` define already as you have for many other aspects of this controller.

Reply:
No `CFG_FOO` available from what I can see. Will add a define.